synapse/util/caches/descriptors.py

```diff
@@ -60,7 +64,12 @@ class _CachedFunction(Generic[F]):
 
 
 class _CacheDescriptorBase:
-    def __init__(self, orig: Callable[..., Any], num_args, cache_context=False):
+    def __init__(
+        self,
+        orig: Callable[..., Any],
```

Comment:
I'm not sure there's anything better to have as the wrapped functions here so I kept it super generic. If there are thoughts of how to improve this, let me know!

Reply:
Could use a `TypeVar` bound to `Callable[..., Any]` if you want to refer to a specific function signature in multiple places throughout the class. I'm not sure if that's actually needed here---maybe the return type of `__get__`? If so, I think you'd have to then make the class generic over that TypeVar.

Reply:
I think the issue with that is that we wouldn't know what to give as the type when instantiating the class since this is all very very abstract code.

Reply:
Agreed, let's leave this as it is.